### ISSUE TITLE
Private endpoints

### ISSUE THREAD

**Issue by itowlson** [fermyon/spin#2416]
One of the follow-on ideas for local service chaining was to allow applications to have "private endpoints" - that is, HTTP components that could be reached _only_ through service chaining and would never be routed.

The first detail that springs to mind is the UI for this.  Currently the HTTP trigger requires a `route` string in `spin.toml`.  Possible ideas:

* Make `route` optional.  The risk here is that if a developer forgets or inadvertently deletes it, then their application behaves unexpectedly and it's not obvious what's gone wrong.  On the other hand there are lots of other optional things which if you delete them things go wrong so maybe that's okay!

* Use a magic value, e.g. `route = "no! no! no!"`.  (Actual magic value may vary.)  This requires no changes to the schema and is easily handled during the existing "parse the routes" step.  On the other hand, magic value.

* Use a structure, e.g. `route = { routable = false}`.  Very explicit and non-magical, not something you're likely to do by accident, but a more substantial change to the schema (with minor implications for lockfile parsing, but I don't think those will be a big deal. He said with doomed optimism), and a bit wordy to type.

Or maybe there's no trigger at all, just a bare component, but then there's no way for Spin to know that it needs to be mapped into the HTTP trigger...

Thoughts, other issues people envision, demands for a SIP, etc.?

**Comment by lann** [comment]
Another option - which I haven't thought through thoroughly - would be to not require _any_ trigger config for private endpoints. This would require a bit of hackery under the current `spin-trigger` design, e.g. synthesizing an http trigger for any component that doesn't already have one.

Regardless of whether it is a synthetic or organic (?) trigger, how about `route = false`, cleverly taking advantage of "route"'s grammatical homonymy.

**Comment by itowlson** [comment]
Thanks - I thought about that the moment after hitting send - edited it in but edits are non-obvious, sorry.  My concern was how do we know that untriggered components are HTTP, but we could make that rule for sure.

`route = false` is nice - thank you!
